src/haiway/files/memory.py

```
@@ -0,0 +1,270 @@
+    ) -> TextIO | BinaryIO:
+        """Create a file handle for reading."""
+        if normalized not in self._files:
+            raise FileNotFoundError(f"File not found: {normalized}")
+
+        raw_data = self._files[normalized]
+
+        if "b" in mode:
+            return BytesIO(raw_data)
+        else:
+            text_data = raw_data.decode(file_encoding or "utf-8")
+            return StringIO(text_data)
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)